packages/flutter/lib/src/gestures/velocity_tracker.dart

```diff
@@ -208,6 +208,34 @@ class _LeastSquaresVelocityTrackerStrategy extends _VelocityTrackerStrategy {
 
 }
 
+/// A velocity in two dimensions.
+class Velocity {
+  const Velocity({ this.pixelsPerSecond });
+
+  /// A velocity that isn't moving at all.
+  static const Velocity zero = const Velocity(pixelsPerSecond: Offset.zero);
+
+  /// The number of pixels per second of velocity in the x and y directions.
+  final Offset pixelsPerSecond;
+
+  /// The square of the magnitude of the velocity (in pixels per second) squared.
+  double get speedSquared {
+    return pixelsPerSecond.dx * pixelsPerSecond.dx
+         + pixelsPerSecond.dy * pixelsPerSecond.dy;
+  }
+
+  bool operator ==(dynamic other) {
+    if (other is! Velocity)
+      return false;
+    final Velocity typedOther = other;
+    return pixelsPerSecond == typedOther.pixelsPerSecond;
+  }
+
+  int get hashCode => pixelsPerSecond.hashCode;
+
+  String toString() => 'Velocity(pixelsPerSecond: $pixelsPerSecond)';
```

Comment:
that's pretty verbose for a primitive-ish type. How about `'Velocity(${pixelsPerSecond.dx.toStringAsFixed(1)}, ${pixelsPerSecond.dy.toStringAsFixed(1)})'`?